src/box/errcode.h

```diff
@@ -115,286 +115,286 @@ struct errcode_record {
+	_(ER_ROLE_LOOP, 87,			"Granting role would create a loop", "role", STRING, "grantee", STRING) \
+	_(ER_GRANT, 88,				"Granting failed", "details", STRING) \
+	_(ER_PRIV_GRANTED, 89,			"Already granted", "user", STRING, "privilege", STRING, "object_type", STRING, "object", STRING) \
+	_(ER_ROLE_GRANTED, 90,			"User already has role", "user", STRING, "role", STRING) \
+	_(ER_PRIV_NOT_GRANTED, 91,		"No such grant", "user", STRING, "privilege", STRING, "object_type", STRING, "object", STRING) \
+	_(ER_ROLE_NOT_GRANTED, 92,		"User does not have role", "user", STRING, "role", STRING) \
 	_(ER_MISSING_SNAPSHOT, 93,		"Can't find snapshot") \
-	_(ER_CANT_UPDATE_PRIMARY_KEY, 94,	"Attempt to modify a tuple field which is part of primary index in space '%s'") \
-	_(ER_UPDATE_INTEGER_OVERFLOW, 95,	"Integer overflow when performing '%c' operation on field %s") \
+	_(ER_CANT_UPDATE_PRIMARY_KEY, 94,	"Attempt to modify a tuple field which is part of primary index", "space", STRING) \
+	_(ER_UPDATE_INTEGER_OVERFLOW, 95,	"Integer overflow", "operation", STRING, "field", STRING) \
 	_(ER_GUEST_USER_PASSWORD, 96,		"Setting password for guest user has no effect") \
 	_(ER_TRANSACTION_CONFLICT, 97,		"Transaction has been aborted by conflict") \
-	_(ER_UNSUPPORTED_PRIV, 98,		"Unsupported %s privilege '%s'") \
-	_(ER_LOAD_FUNCTION, 99,			"Failed to dynamically load function '%s': %s") \
-	_(ER_FUNCTION_LANGUAGE, 100,		"Unsupported language '%s' specified for function '%s'") \
-	_(ER_RTREE_RECT, 101,			"RTree: %s must be an array with %u (point) or %u (rectangle/box) numeric coordinates") \
-	_(ER_PROC_C, 102,			"%s") \
-	_(ER_UNKNOWN_RTREE_INDEX_DISTANCE_TYPE, 103, "Unknown RTREE index distance type %s") \
-	_(ER_PROTOCOL, 104,			"%s") \
-	_(ER_UPSERT_UNIQUE_SECONDARY_KEY, 105,	"Space %s has a unique secondary index and does not support UPSERT") \
-	_(ER_WRONG_INDEX_RECORD, 106,		"Wrong record in _index space: got {%s}, expected {%s}") \
-	_(ER_WRONG_INDEX_PARTS, 107,		"Wrong index part %u: %s") \
-	_(ER_WRONG_INDEX_OPTIONS, 108,		"Wrong index options: %s") \
-	_(ER_WRONG_SCHEMA_VERSION, 109,		"Wrong schema version, current: %d, in request: %llu") \
-	_(ER_MEMTX_MAX_TUPLE_SIZE, 110,		"Failed to allocate %u bytes for tuple: tuple is too large. Check 'memtx_max_tuple_size' configuration option.") \
-	_(ER_WRONG_SPACE_OPTIONS, 111,		"Wrong space options: %s") \
-	_(ER_UNSUPPORTED_INDEX_FEATURE, 112,	"Index '%s' (%s) of space '%s' (%s) does not support %s") \
-	_(ER_VIEW_IS_RO, 113,			"View '%s' is read-only") \
+	_(ER_UNSUPPORTED_PRIV, 98,		"Unsupported privilege", "object_type", STRING, "privilege", STRING) \
+	_(ER_LOAD_FUNCTION, 99,			"Failed to dynamically load function", "function", STRING, "details", STRING) \
+	_(ER_FUNCTION_LANGUAGE, 100,		"Unsupported function language", "language", STRING, "function", STRING) \
+	_(ER_RTREE_RECT, 101,			"Coordinates number does not make a point or rectangle", "object", STRING, "dimension", UINT) \
```

Comment:
Bad message. Unclear that it's for the rtree index.

Reply:
Fixed as discussed f2f.

<details>
<summary>diff</summary>

```diff
diff --git a/src/box/errcode.h b/src/box/errcode.h
index 5cc7a738e..34c64cd0d 100644
--- a/src/box/errcode.h
+++ b/src/box/errcode.h
@@ -215,7 +215,8 @@ struct errcode_record {
 	_(ER_UNSUPPORTED_PRIV, 98,		"Unsupported privilege", "object_type", STRING, "privilege", STRING) \
 	_(ER_LOAD_FUNCTION, 99,			"Failed to dynamically load function", "func", STRING, "details", STRING) \
 	_(ER_FUNCTION_LANGUAGE, 100,		"Unsupported function language", "value", STRING, "func", STRING) \
-	_(ER_RTREE_RECT, 101,			"Coordinates number does not make a point or rectangle", "object", STRING, "dimension", UINT) \
+	_(ER_RTREE_RECT, 101,			"Key/field of RTREE index should be array with dimension (point) or 2 * dimension (rectangle) coordinates", \
+						"dimension", UINT) \
 	_(ER_PROC_C, 102,			"Error in C function") \
 	/* ER_UNKNOWN_RTREE_INDEX_DISTANCE_TYPE, 103, Unused */ \
 	_(ER_PROTOCOL, 104,			"Protocol error", "details", STRING) \
diff --git a/src/box/index.cc b/src/box/index.cc
index 339dc11d7..7b9de6b9d 100644
--- a/src/box/index.cc
+++ b/src/box/index.cc
@@ -100,7 +100,7 @@ key_validate(const struct index_def *index_def, enum iterator_type type,
 				return -1;
 			uint32_t array_size = mp_decode_array(&key);
 			if (array_size != d && array_size != d * 2) {
-				diag_set(ClientError, ER_RTREE_RECT, "Key", d);
+				diag_set(ClientError, ER_RTREE_RECT, d);
 				return -1;
 			}
 			for (uint32_t part = 0; part < array_size; part++) {
diff --git a/src/box/memtx_rtree.cc b/src/box/memtx_rtree.cc
index 061db1ed5..066e02368 100644
--- a/src/box/memtx_rtree.cc
+++ b/src/box/memtx_rtree.cc
@@ -73,7 +73,7 @@ mp_decode_num(const char **data, uint32_t fieldno, double *ret)
  */
 static inline int
 mp_decode_rect(struct rtree_rect *rect, unsigned dimension,
-	       const char *mp, unsigned count, const char *what)
+	       const char *mp, unsigned count)
 {
 	coord_t c = 0;
 	if (count == dimension) { /* point */
@@ -95,7 +95,7 @@ mp_decode_rect(struct rtree_rect *rect, unsigned dimension,
 			rect->coords[i * 2 + 1] = c;
 		}
 	} else {
-		diag_set(ClientError, ER_RTREE_RECT, what, dimension);
+		diag_set(ClientError, ER_RTREE_RECT, dimension);
 		return -1;
 	}
 	rtree_rect_normalize(rect, dimension);
@@ -115,7 +115,7 @@ mp_decode_rect_from_key(struct rtree_rect *rect, unsigned dimension,
 {
 	if (part_count == 1)
 		part_count = mp_decode_array(&mp);
-	return mp_decode_rect(rect, dimension, mp, part_count, "Key");
+	return mp_decode_rect(rect, dimension, mp, part_count);
 }
 
 static inline int
@@ -128,7 +128,7 @@ extract_rectangle(struct rtree_rect *rect, struct tuple *tuple,
 				index_def->key_def->parts, MULTIKEY_NONE);
 	unsigned dimension = index_def->opts.dimension;
 	uint32_t count = mp_decode_array(&elems);
-	return mp_decode_rect(rect, dimension, elems, count, "Field");
+	return mp_decode_rect(rect, dimension, elems, count);
 }
 /* {{{ MemtxRTree Iterators ****************************************/
 

```

</details>